### PR TITLE
Use AES-GCM for encryption

### PR DESCRIPTION
**Pull request by orhoj**
## Purpose
Use AES-GCM for seed phrase encryption for proper checks of whether or not decryption went well. Uses browser libraries for decryption/encryption instead cryptojs.

NB: This is not compatible with existing seed phrases already in storage. Recover from a clean wallet.

## Changes
- Exchange encryption algorithm.
- Updated some components and atom usage as decryption and encryption is now `async`.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: packages/browser-wallet/src/popup/shared/utils/account-helpers.ts [hunk 1]
```diff
@@ -49,11 +49,11 @@ export function usePrivateKey(accountAddress: string | undefined): string | unde
...
-        if (!accountAddress || !credential || !identity) {
+        if (!accountAddress || !credential || !identity || seedPhrase.state !== 'hasData') {
             return undefined;
         }
 
-        return ConcordiumHdWallet.fromHex(seedPhrase, getNet(network))
+        return ConcordiumHdWallet.fromHex(seedPhrase.data, getNet(network))
             .getAccountSigningKey(identity.providerIndex, identity.index, credential.credNumber)
             .toString('hex');
     }, [credential?.credId, seedPhrase, identity?.index]);
```
Comment (shjortConcordium) [inline packages/browser-wallet/src/popup/shared/utils/account-helpers.ts:59]:
```suggestion
    }, [credential?.credId, seedPhrase.state, identity?.index]);
```
oder was?

Comment (orhoj) [reply packages/browser-wallet/src/popup/shared/utils/account-helpers.ts:59]: The value is present in the `data` subfield when `state === 'hasData'`.

Comment (shjortConcordium) [reply packages/browser-wallet/src/popup/shared/utils/account-helpers.ts:59]: Sure.
But don't we have problems with having objects in the dependency array?
And the assumption is that the data won't change?
Maybe this is fine actually

**Review by shjortConcordium — APPROVED** [review]

File: packages/browser-wallet/src/popup/shared/utils/account-helpers.ts (hunk 1 above)
Comment (orhoj) [reply packages/browser-wallet/src/popup/shared/utils/account-helpers.ts:59]: I misread where the comment was placed. I will change it to state, and yes the assumption is that data won't change here.

**Pull request merged**